agent/src/api/v1/jobs.rs

```diff
@@ -1,82 +1,55 @@
-use std::{collections::HashMap, path::PathBuf};
-
-use anyhow::Result;
 use axum::{
-    extract::{Multipart, Path, State},
+    body::Bytes,
+    extract::{Path, State},
     http::StatusCode,
-    response::{IntoResponse, Response},
+    response::IntoResponse,
     Json,
 };
 use engine::{errors::ServalEngineError, ServalEngine};
-use utils::structs::WasmResult;
+use utils::structs::Job;
 
 use crate::structures::*;
 
-/// This is the main worker endpoint. It accepts incoming jobs and runs them.
-pub async fn incoming(state: State<AppState>, mut multipart: Multipart) -> Response {
-    let mut envelope: Option<Envelope> = None;
-    let mut binary: Option<Vec<u8>> = None;
-    let mut input: Option<Vec<u8>> = None;
+/// Report on runtime history
+pub async fn monitor_status(state: State<AppState>) -> Json<RunnerState> {
+    let state = state.lock().await;
+    Json(state.clone())
+}
 
-    // chomp up form input here
-    while let Some(field) = multipart.next_field().await.unwrap() {
-        let name = field.name().unwrap().to_string();
-        let data = field.bytes().await.unwrap();
-        match name.as_str() {
-            "envelope" => {
-                let data = data.to_vec();
-                let Ok(parsed) = serde_json::from_slice(&data) else {
-                    // this is not good enough
-                    return (StatusCode::BAD_REQUEST, "job envelope is invalid".to_string()).into_response();
-                };
-                envelope = Some(parsed);
-            }
-            "input" => {
-                input = Some(data.to_vec());
-            }
-            "executable" => {
-                binary = Some(data.to_vec());
-            }
-            _ => {
-                log::info!("ignoring unknown field `{name}`");
-            }
-        }
-    }
+/// Get running jobs
+pub async fn running(_state: State<AppState>) -> impl IntoResponse {
+    StatusCode::NOT_IMPLEMENTED
+}
+
+/// This is the main worker endpoint. It accepts incoming jobs and runs them.
+pub async fn run_job(
+    Path(name): Path<String>,
+    state: State<AppState>,
+    input: Bytes,
+) -> impl IntoResponse {
+    let mut lock = state.lock().await;
+    let storage = lock.storage.as_ref().unwrap();
+    let Ok(manifest) = storage.manifest(&name).await else {
```

Comment:
Does this mean that only nodes that have storage locally available can run jobs?

Reply:
Makes sense. Thoughts on swapping `BlobStore` out for an abstract trait that has `CacacheBlobStore` as one implementation and `RemoteBlobStore` (something that proxies to a serval_storage instance) as the other? If that sounds reasonable to me I am happy to start chipping away at it.